logos/src/logos/capacity/capacity_planner.py

```
@@ -923,6 +972,17 @@ def _lane_is_in_load_cooldown(
         check_time = time.time() if now is None else now
         return (check_time - loaded_at) < self._load_cooldown_seconds
 
+    def _model_has_cold_queued_waiter(self, provider_id: int, model_name: str) -> bool:
+        """Return True if any queued entry for this model on this provider was
+        flagged is_cold_at_queue at enqueue. Drives extended lane tenure so a
+        freshly-woken lane isn't put back to sleep before serving the request
+        that triggered its wake.
+        """
+        try:
+            return self._facade.has_cold_queued_entries_by_model_name(model_name, provider_id)
+        except Exception:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)